src/material/tabs/testing/tab-group-harness.ts

```diff
@@ -52,4 +50,13 @@ export class MatTabGroupHarness extends ComponentHarness {
     }
     throw new Error('No selected tab could be found.');
   }
+
+  /** Selects a tab in this tab group. */
+  async selectTab(filter: TabHarnessFilters = {}): Promise<void> {
+    const tabs = await this.getTabs(filter);
+    if (!tabs.length) {
+      throw Error(`Cannot find mat-tab matching ${JSON.stringify(filter)}`);
```

Comment:
Optional suggestion: Mainly because what if no filter is specified? it will be `Cannot find mat-tab matching {}`. That is most likely confusing. 

```suggestion
      throw Error(`Cannot find mat-tab matching filter: ${JSON.stringify(filter)}`);
```